test/Tests/src/Semantic/Instrumentor_Spec.enso

```diff
@@ -0,0 +1,60 @@
+from Standard.Base import all
+from Standard.Test import all
+
+fib n = if n <= 1 then 1 else
+    a = fib n-1
+    b = fib n-2
+    a+b
+
+spec =
+    Test.group "Instrument fibonacci" <|
+        Test.specify "collect and filter on return updates" <|
+            b = Vector.new_builder
+
+            collect uuid:Text result =
+                a_plus_b_uuid = "00000000-aaaa-bbbb-0000-000000000000" # UUID for a+b
+                if uuid == a_plus_b_uuid then
+                    b.append result
+                Nothing
+
+            instrumenter = Meta.meta .fib . instrument . on_return collect . activate
+
+            instrumenter . with _->
+                result = fib 10
+
+                v = b.to_vector
+
+                v.length . should_equal 1
+                v.at 0 . should_equal 89
+                result . should_equal 89
+
+            instrumenter.finalize
+
+            result = fib 10
+            result . should_equal 89
+
+            # no more instrumenting after finalize
+            b.to_vector.length . should_equal 1
+
+        Test.specify "replay with caches and specify different result" <|
+            replay uuid:Text = case uuid of
+                "00000000-ffff-bbbb-0000-000000000000" -> 42
+                _ -> Nothing
+
+            instrumenter = Meta.meta .fib . instrument . on_enter replay . activate
+
+            instrumenter . with _->
+                result = fib 10
+                result . should_equal 42
+
+            instrumenter.finalize
```

Comment:
IMHO doing a block like this will work better.

`Managed_Resource` is for resources that we may need to e.g. pass around.

At least looking at the use-case in these tests it seems that it should be enough to just be able to run a block of code with the instrumenter.

If we need to pass something around, we can just pass around the builder and create the instrumenter for each piece, right?

Am I missing something here?

Reply:
I think we need to _pass the instrumenter around_. If we want to use it for visualizations, then the IDE would send a message, activate the instrumenter and let it remain activated. The usage in block is nice in tests, but real usage will likely require long term activation.

Reply:
I see. I think I'm not aware of what are the planned use cases then to be able to have a good mind map of it, but I trust your choice.